env/CMakeLists.txt

```diff
@@ -17,14 +17,17 @@ if (NOT EXISTS "${TTMLIR_TOOLCHAIN_DIR}")
     message( FATAL_ERROR "The directory ${TTMLIR_TOOLCHAIN_DIR} does not exist. Please create it before running this script.\n  sudo mkdir -p ${TTMLIR_TOOLCHAIN_DIR}\n  sudo chown -R $ENV{USER} ${TTMLIR_TOOLCHAIN_DIR}")
 endif()
 
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Build type" FORCE)
```

Comment:
Hmm, I feel like the default for env should be MinSizeRel. Most users are probably not going to need/want debug symbols in llvm.

Reply:
@vmilosevic Maybe toolchain can remain MinSizeRel if we have dev image where its prebuilt with debug?

Reply:
Here are LLVM suggestions:
<img width="1041" alt="image" src="https://github.com/user-attachments/assets/57b212f4-6ffd-4cff-86a7-39844be4075d">

Reply:
If I understand correctly, you want tt-mlir relese build to have LLVM build with MinSizeRel, 
and tt-mlir debug build to build LLVM with debug config?

We can run CI for both debug and release, and it will make both types of images.

What should be the default config for tt-mlir, when CMAKE_BUILD_TYPE is not specified? IMHO it make sense to be debug, but we can set anything you like

Reply:
I agree default TT-MLIR build should be debug. What Nick wants is that even for debug TT-MLIR, LLVM toolchain to be MinSizeRel. Now my thinking was if I can pick an image where LLVM is prebuilt to be Debug, Im fine with that if not I think it should at least be RelWithDebInfo.

Reply:
So this is just the default we have in the cmake file, ie what someone gets if they manually build the env and the project. This is a separate default to what we choose to build in our docker image.

- Docker: let's build debug as it's configured on this PR
- Cmake default: should probably be MinSizeRel or MinSizeRelWithDebug.